recipe/activate.sh

```diff
@@ -1,7 +1,45 @@
 #!/bin/bash
 
-if [[ -n "$CXX" ]]; then
-    export NVCC_PREPEND_FLAGS_BACKUP="${NVCC_PREPEND_FLAGS}"
-    export NVCC_PREPEND_FLAGS="${NVCC_PREPEND_FLAGS} -ccbin=${CXX}"
+if [[ ! -z "${CFLAGS+x}" ]]
+then
+  export CFLAGS_CUDA_NVCC_BACKUP="${CFLAGS}"
 fi
-if [ -z "${CXX+x}" ]; then echo 'cuda-nvcc: Please add the `c-compiler` and `cxx-compiler` packages to the environment.'; fi
+
+if [[ ! -z "${CPPFLAGS+x}" ]]
+then
+  export CPPFLAGS_CUDA_NVCC_BACKUP="${CPPFLAGS}"
+fi
+
+if [[ ! -z "${CXXFLAGS+x}" ]]
+then
+  export CXXFLAGS_CUDA_NVCC_BACKUP="${CXXFLAGS}"
+fi
+
+if [[ ! -z "${NVCC_PREPEND_FLAGS +x}" ]]
```

Comment:
Sorry a space crept in here 🤦‍♂️

```suggestion
if [[ ! -z "${NVCC_PREPEND_FLAGS+x}" ]]
```